apps/passport-server/src/util/telegramHelpers.ts

```diff
@@ -18,6 +36,54 @@ export interface SessionData {
 
 export type BotContext = Context & SessionFlavor<SessionData>;
 
+function isFulfilled<T>(
+  promiseSettledResult: PromiseSettledResult<T>
+): promiseSettledResult is PromiseFulfilledResult<T> {
+  return promiseSettledResult.status === "fulfilled";
+}
+
+/**
+ * Fetches the chat object for a list contaning a telegram chat id
+ */
+export const getEventsWithChats = async <
+  T extends LinkedPretixTelegramEvent | ChatIDWithEvents
+>(
+  db: Pool,
+  ctx: BotContext,
+  chats: T[]
+): Promise<EventWithChat<T>[]> => {
+  const eventsWithChatsRequests = chats.map(async (e) => {
```

Comment:
I'm a bit confused in regards to the naming of this function - where do "events" come in? Seems like it's just fetching chat objects for a list of IDs

Reply:
Makes sense, i updated the naming a bit to 
```ts
  db: Pool,
  ctx: BotContext,
  chats: T[]
): Promise<ChatIDWithChat<T>[]> => {
  const chatIDsToChatRequests = chats.map(async (e) => {
  ```